frontend/src/screens/apps/Index.tsx

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable react-refresh/only-export-components */
+import { useEffect, useState } from "react";
+import { useNavigate } from 'react-router-dom';
+
+import Loading from '../../components/Loading';
+import { App, NostrEvent, ListAppsResponse } from "../../types";
+
+function Connections() {
+  const [apps, setApps] = useState<App[] | null>(null);
+  const [eventsCounts, setEventsCounts] = useState<Record<App["id"], number> | null>(null);
+  const [lastEvents, setLastEvents] = useState<Record<App["id"], NostrEvent> | null>(null);
+  const navigate = useNavigate();
+
+  const handleRowClick = (appId: App["nostrPubkey"]) => {
+    navigate(`/apps/${appId}`);
+  };
+
+  useEffect(() => {
+    const fetchAppsData = async () => {
+      try {
+        const response = await fetch("/api/apps");
+        const data: ListAppsResponse = await response.json();
+        setApps(data.apps);
+        setEventsCounts(data.eventsCounts);
+        setLastEvents(data.lastEvent);
+      } catch (error) {
+        // TODO: Show error page
+        navigate("/apps?q=notfound");
+        console.error('Error fetching app data:', error);
+      }
+    };
+
+    fetchAppsData();
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return(
+    <>
+      <div className="mb-4 flex justify-between items-center">
+        <h2 className="font-bold text-2xl font-headline dark:text-white">Connected apps</h2>
+        <a className="inline-flex bg-purple-700 cursor-pointer dark:text-neutral-200 duration-150 focus-visible:ring-2 focus-visible:ring-offset-2 focus:outline-none font-medium items-center justify-center px-3 md:px-6 py-2 md:py-3 rounded-lg shadow text-white transition {{if not .User}}opacity-50{{else}}hover:bg-purple-900{{end}} text-sm md:text-base" href="/apps/new">
```

Comment:
the amount of classes are huge. Maybe we can improve this in a follow-up PR by creating some UI components.